test/libsolidity/util/TestFileParser.cpp

```diff
@@ -169,6 +171,23 @@ vector<solidity::frontend::test::FunctionCall> TestFileParser::parseFunctionCall
 	return calls;
 }
 
+vector<string> TestFileParser::parseFunctionCallSideEffects()
+{
+	vector<string> result;
+	while (accept(Token::Tilde, false))
+	{
+		string effect = m_scanner.currentLiteral();
+		result.emplace_back(effect);
+
+		if (m_scanner.currentToken() == Token::Tilde)
+			m_scanner.scanNextToken();
```

Comment:
I think that the token must be a tilde at this point, right? `while` checks for it and I don't think `currentLiteral()` can change the token.

```suggestion
		soltestAssert(m_scanner.currentToken() == Token::Tilde, "");
		m_scanner.scanNextToken();
```